### ISSUE TITLE
[Accessibility] Increase minimum font size to 1 rem

### ISSUE THREAD

**Issue by phoeinx** [tactilenews/100eyes#1444]
I don't like the idea of text being smaller than `1rem/16px` because this is generally accepted as the minimum font-size for `a11y`. See [Minimum font size?](https://accessibleweb.com/question-answer/minimum-font-size/)

> While there is no official minimum font size for the web, it is generally agreed upon that 16px for body text is a good starting point.

I personally would like to scale up the font-size to meet this suggestion, but not sure how others feel.

_Originally posted by @mattwr18 in https://github.com/tactilenews/100eyes/pull/1419#discussion_r966038470_

**Comment by FrauCsu** [comment]
If 1rem/16px is recommended, then let's adapt. Or why not, @drjakob ?

**Comment by phoeinx** [comment]
I had a quick deeper look into this and I think we might have already resolved it by giving the button a font size of `var(--font-size-s)`.

`--font-size-s` is defined as 0.85 **em** and therefore dependent on the font-size of its parent element. 
For the button which we discussed the parent element has a font-size of `--font-size` which in our case evaluates to `19,36px` and the resulting font size of the button is `19,36 * 0.85 = 16.456px`.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/34538290/190194647-804c011c-0962-4006-83de-784412d4ddf9.png">

However, our other commonly used font-size `--font-size-xs`,defined as 0.75em, results in font sizes smaller than 16px, e.g. in our footer which has a font-size of 14.52px:

<img width="608" alt="image" src="https://user-images.githubusercontent.com/34538290/190195416-d7415a58-8931-4dd3-93ba-c8ac6f96a1af.png">

I think we can assume that we will not nest `--font-size` attributes and that therefore `--font-size-xs` with `0.75em` will be the smallest font-size with then `19.36px * 0,75 = 14.52px` as can be seen in the footer.

If we want to maintain the relative proportions of our font sizes while also making sure that each text has at least a font size of 16px the easiest solution would be to increase `--font-size` to around 21px. 

--- 

Here are two screenshots with the different viewports I get in Chrome on MacOS on 100% Zoom with:


**A: Current Version:**
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/34538290/190242302-33ee4b59-57c5-443a-887a-1f6ca3af4482.png">


**B: Bigger font sizes -> Base font size having `21px`:**
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/34538290/190198102-3b9d198d-163f-452f-9c9b-1b92b4cb47d3.png">

--- 

For me both options would be fine. However, reading the [Minimum font size?](https://accessibleweb.com/question-answer/minimum-font-size/) documentation from Accessible Web I think our current sizes are actually already acceptable:

> Of course, some text will be smaller and headings will often times be larger. But, the main body text (like what you're reading right now) should usually be 16px or larger. Additionally, it is officially stated within the WCAG that visitors must be able to resize all text on the page to at least 200% of its original size.

As far as I saw `--font-size-xs` is only used for the chat preview text and footer and tab input, so all the main-ish body text is at least `16px`. I also just checked that one can use our app on 200% Zoom, which works well (and except for the admin dashboard without horizontal scrolling).

--- 

As a side note: I am kind of confused as to why `var(--font-size)` is `19.36px` in our case, given that it gets defined as `1.1rem` in `variables.css`. Assuming that `1rem` equals `16px`, `1.1rem` should be `17.6px`, right? I wasn't able to quickly identify what happens here, @tillprochaska @mattwr18 do you know why it is like that? (I checked that on other webpages in my browser `1rem` evaluates to `16px`.

**Comment by tillprochaska** [comment]
> As a side note: I am kind of confused as to why var(--font-size) is 19.36px in our case, given that it gets defined as 1.1rem in variables.css. Assuming that 1rem equals 16px, 1.1rem should be 17.6px, right? I wasn't able to quickly identify what happens here, @

I think that is because we have the following global style:

```
html, body {
  font-size: 1.1rem;
}
```

Which means that the root font size will be 1.1rem = 17.6px. The body font size is also set to 1.1rem (and as it’s using rems, that is relative to the root/html element font size), i.e. the body font size is 1.1 * 17.6px = 19.36px.

I admit this isn’t super clean, although I don’t think it has any significant disadvantages except for being slightly confusing.

**Comment by phoeinx** [comment]
Hi @tillprochaska, Thanks for the quick reply! Makes a lot of sense.

**Comment by mattwr18** [comment]
Thanks for looking into this @phoeinx! I had noticed that the computed font-size was larger than the declared font-size, but didn't look into why. I agree that it's ok to have some text that is slightly smaller than the recommended minimum cause as you pointed out, it's for main body text and the important thing is that we add our font-size in a way that allows users to change their base font-size and it be scaled up/down accordingly, or to zoom. We already do this by using `rem`.

@tillprochaska it doesn't really make sense to me why we would want to declare our base size as larger than the most browsers' default font-size of 16px. I also don't believe we should set the `root` font-size in `rem`, but rather in px and all other declarations in `rem` so that they scale according. See [Font Size Idea: px at the Root, rem for Components, em for Text Elements](https://css-tricks.com/rems-ems/) and [Is the default font-size of every browser 16px? Why?](https://stackoverflow.com/questions/29511983/is-the-default-font-size-of-every-browser-16px-why)

As far as I can see, this is not an `a11y` issue as I had imagined it, it is more an issue of it being confusing. I would suggest updating these in a way that the declared values would equal the computed values.

**Comment by phoeinx** [comment]
I agree that it would be nice to make the font-size calculation easier to understand. As far as I understood it this should be as  easy as setting the root font-size to ~20px (and not setting the font size for the HTML tag), right?

One question I would have there is if setting the root font size in `px` would overwrite the font size setting in browsers where users can adjust the root font size to their needs. See [Set page or font size for all webpages](https://support.google.com/chrome/answer/96810?hl=en&co=GENIE.Platform%3DDesktop)

**Comment by mattwr18** [comment]
I'm closing this as it's not an `a11y` issue actually, it was just confusing. I attempted a quick fix to simplify it, but we rely on the font-size declaration in `html/body` as a fallback for lots of places where we don't have a declared font-size, so we would need to go through the entire application and declare them.

At the moment, it's hard to justify spending dev time on this.
